analyzers/analyzer.go

```diff
@@ -60,7 +61,13 @@ func New(m module.Module) (Analyzer, error) {
 	case pkg.Maven:
 		return maven.New(m)
 	case pkg.NodeJS:
-		return nodejs.New(m)
+		sysNpmTool, err := npm.New(m.Options)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return nodejs.New(m, sysNpmTool)
```

Comment:
You don't need to make the tool an argument in `nodejs.New`. We can directly set the `Tool` field in the test instead.